nexong/models.py

```diff
@@ -210,61 +237,83 @@ class Lesson(models.Model):
+
+
+class EvaluationType(models.Model):
+    name = models.CharField(max_length=100)
+    description = models.CharField(max_length=1000, null=True, blank=True)
     evaluation_type = models.CharField(
         max_length=10, choices=EVALUATION_TYPE, default=DAILY
     )
-    student = models.ForeignKey(
-        Student, on_delete=models.CASCADE, related_name="student_evaluations"
+    grade_system = models.CharField(
+        max_length=20, choices=GRADESYSTEM, default=ZERO_TO_TEN
     )
     lesson = models.ForeignKey(
         Lesson, on_delete=models.CASCADE, related_name="student_evaluations"
     )
 
 
-class LessonEvaluation(models.Model):
-    name = models.CharField(max_length=100)
-    description = models.CharField(max_length=1000)
-    grade_system = models.CharField(
-        max_length=20, choices=GRADESYSTEM, default=ZERO_TO_TEN
-    )
+class StudentEvaluation(models.Model):
     grade = models.IntegerField(
         validators=[MinValueValidator(0), MaxValueValidator(10)], default=0
     )
     date = models.DateField()
-    evaluation_type = models.CharField(
-        max_length=10, choices=EVALUATION_TYPE, default=DAILY
+    comment = models.CharField(max_length=1000, null=True, blank=True)
+    student = models.ForeignKey(
+        Student, on_delete=models.CASCADE, related_name="student_evaluations"
+    )
+    evaluation_type = models.ForeignKey(
+        EvaluationType, on_delete=models.CASCADE, related_name="student_evaluations"
     )
+
+
+class LessonAttendance(models.Model):
+    date = models.DateField()
     lesson = models.ForeignKey(
-        Lesson, on_delete=models.CASCADE, related_name="lesson_evaluations"
+        Lesson, on_delete=models.CASCADE, related_name="lesson_attendances"
     )
-    family = models.ForeignKey(
-        Family, on_delete=models.CASCADE, related_name="lesson_evaluations"
+    volunteer = models.ForeignKey(
+        Volunteer, on_delete=models.CASCADE, related_name="lesson_attendances"
     )
 
 
+class LessonEvent(models.Model):
+    name = models.CharField(max_length=100)
+    description = models.CharField(max_length=1000)
+    place = models.CharField(max_length=1000)
+    max_volunteers = models.IntegerField(validators=[MinValueValidator(0)])
+    start_date = models.DateTimeField()
+    end_date = models.DateTimeField()
+    lesson = models.ForeignKey(Lesson, on_delete=models.CASCADE, null=True, blank=True)
+    price = models.IntegerField(validators=[MinValueValidator(0)], default=0)
+    educators = models.ManyToManyField(Educator, related_name="lesson_events")
+    attendees = models.ManyToManyField(Student, related_name="lesson_events")
+    volunteers = models.ManyToManyField(Volunteer, related_name="lesson_events")
+
+
 class Event(models.Model):
     name = models.CharField(max_length=100)
     description = models.CharField(max_length=1000)
     place = models.CharField(max_length=1000)
     capacity = models.IntegerField(default=0, validators=[MinValueValidator(0)])
```

Comment:
This is a mistake in the DB, sorry 😑. Please remove capacity. The logic should be like this: attendees <= max_attendees and volunteers <= max_volunteers.